### PR TITLE
Goonhub API: Ban Endpoints

### PR DESCRIPTION
**Pull request by TheTyrant25**
[TOOLING] [INTERNAL]
## About the PR
Adds api support for the ban endpoints. Adds three new request bodies to match.

Adds a template for people in the future trying to set up routes. Also for my own reference.

## why tho
Goonhub API is good.

### CODE REVIEW

File: code/modules/goonhub/api/routes/bans/AddBans.dm [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/// GET /bans
+/// Get
+/datum/apiRoute/bans/add
+	method = RUSTG_HTTP_METHOD_GET
```
Comment (ZeWaka) [inline code/modules/goonhub/api/routes/bans/AddBans.dm:4]:
```suggestion
/// POST /bans
/// Add a ban for given player data
/datum/apiRoute/bans/add
	method = RUSTG_HTTP_METHOD_POST
```

File: code/modules/goonhub/api/routes/bans/AddBanDetails.dm [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/// POST /bans/details/{ban}
+/// Add Details
```
Comment (ZeWaka) [inline code/modules/goonhub/api/routes/bans/AddBanDetails.dm:2]: need to use the descriptions.
```suggestion
/// Add new player details to an existing ban. This should be used when an evasion attempt is detected.
```

**Review by ZeWaka — CHANGES_REQUESTED** [review]
we'll need a way to differentiate between query and path parameters, right now you're missing all the query ones

**Comment by TheTyrant25** [comment]
okay so now everything should be fixed. Except for uh. Path for the bans. Which i don't know what to do with.

File: code/modules/goonhub/api/requests/bans/CheckBansBody.dm [hunk 3]
```diff
@@ -0,0 +1,35 @@
+
+/datum/apiBody/bans/check
+	var/server_id			= "string"
+	var/ckey				= "string"
+	var/comp_id				= "string"
+	var/ip					= "string"
```
Comment (ZeWaka) [inline code/modules/goonhub/api/requests/bans/CheckBansBody.dm:6]:
```suggestion
	var/server_id	= "string"
	var/ckey		= "string"
	var/comp_id		= "string"
	var/ip			= "string"
```

File: code/modules/goonhub/api/requests/bans/CheckBansBody.dm [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		|| isnull(src.ip) \
+	)
+		return FALSE
+
+/datum/apiBody/bans/check/toJson()
+	return json_encode(list(
+		"server_id"			= src.server_id,
+		"ckey"				= src.ckey,
+		"comp_id"			= src.comp_id,
+		"ip"				= src.ip
```
Comment (ZeWaka) [inline code/modules/goonhub/api/requests/bans/CheckBansBody.dm:34]:
```suggestion
		"server_id"	= src.server_id,
		"ckey"		= src.ckey,
		"comp_id"	= src.comp_id,
		"ip"		= src.ip
```

File: code/modules/goonhub/api/requests/bans/AddBansDetailBody.dm [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+		|| isnull(src.comp_id) \
+		|| isnull(src.ip) \
+	)
+		return FALSE
+
+/datum/apiBody/bans/add_detail/toJson()
+	return json_encode(list(
+		"ckey"				= src.ckey,
+		"comp_id"			= src.comp_id,
+		"ip"				= src.ip
```
Comment (ZeWaka) [inline code/modules/goonhub/api/requests/bans/AddBansDetailBody.dm:29]:
```suggestion
		"ckey"		= src.ckey,
		"comp_id"	= src.comp_id,
		"ip"		= src.ip
```

File: code/modules/goonhub/api/requests/bans/AddBansDetailBody.dm [hunk 6]
```diff
@@ -0,0 +1,30 @@
+
+/datum/apiBody/bans/add_detail
+	var/ckey				= "string"
+	var/comp_id				= "string"
+	var/ip					= "string"
```
Comment (ZeWaka) [inline code/modules/goonhub/api/requests/bans/AddBansDetailBody.dm:5]:
```suggestion
	var/ckey	= "string"
	var/comp_id	= "string"
	var/ip		= "string"
```

File: code/modules/goonhub/api/requests/bans/AddBansBody.dm [hunk 7]
```diff
@@ -0,0 +1,55 @@
+
+/datum/apiBody/bans/add
+	var/game_admin_ckey	= "string"
+	var/round_id			= 0
+	var/server_id			= "string"
+	var/ckey				= "string"
+	var/comp_id				= "string"
+	var/ip					= "string"
+	var/reason				= "string"
+	var/duration			= 0
```
Comment (ZeWaka) [inline code/modules/goonhub/api/requests/bans/AddBansBody.dm:10]:
```suggestion
	var/round_id	= 0
	var/server_id	= "string"
	var/ckey		= "string"
	var/comp_id		= "string"
	var/ip			= "string"
	var/reason		= "string"
	var/duration	= 0
```

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**